tests/test_kde.py

```diff
@@ -56,8 +58,50 @@ def test_weighted_samples_2d():
     hist_neglect_weights = kde_histogram_2d(
         x_weighted[:, 0], x_weighted[:, 1], bins1, bins2, sigma
     )
-    print(hist_unweighted[5])
-    print(hist_weighted[5])
-    print(hist_neglect_weights[5])
+
     assert torch.allclose(hist_unweighted, hist_weighted)
     assert not torch.allclose(hist_weighted, hist_neglect_weights)
+
+
+def test_kde_1d_basic_usage():
+    """
+    Test that basic usage of the 1D KDE histogram implementation works, and that the
+    output has the correct shape.
+    """
+    data = torch.randn(100)  # 5 beamline states, 100 particles in 1D
+    bins = torch.linspace(0, 1, 10)  # A single histogram
+    sigma = torch.tensor(0.1)  # A single bandwidth
+
+    pdf = kde_histogram_1d(data, bins, sigma)
+
+    assert pdf.shape == Size([10])  # 5 histograms at 10 points
+
+
+def test_kde_1d_enforce_bins_shape():
+    """
+    Test that the 1D KDE histogram implementation correctly enforces the shape of the
+    bins tensor, i.e. throws an error if the bins tensor has the wrong shape.
+    """
+    data = torch.randn(100)  # 5 beamline states, 100 particles in 1D
+    bins = torch.linspace(0, 1, 10)  # A single histogram
+
+    with pytest.raises(ValueError):
+        kde_histogram_1d(data, bins, torch.rand(3) + 0.1)
```

Comment:
I'm a bit confused about what the "5 beamline states" comment refers to.

Reply:
It seems the vector dimension was intended but missing. Now it makes sense, right?